ipaserver/plugins/passwd.py

```diff
@@ -147,3 +148,78 @@ def execute(self, principal, password, current_password, **options):
             result=True,
             value=principal,
         )
+
+
+@register()
+class passwd_generate(Command):
+    __doc__ = _("Autogenerate a password.")
+
+    takes_options = (
+        Int('uppercase',
+            label=_('Uppercase'),
+            doc=_('Number of uppercase characters'),
+            required=False,
+        ),
+        Int('lowercase',
+            label=_('Lowercase'),
+            doc=_('Number of lowercase characters'),
+            required=False,
+        ),
+        Int('digits',
+            label=_('Digits'),
+            doc=_('Number of digits'),
+            required=False,
+        ),
+        Int('special',
+            label=_('Special characters'),
+            doc=_('Number of special characters'),
+            required=False,
+        ),
+        Int('length',
+            label=_('Length'),
+            doc=_('Password Length'),
+            required=False,
+        ),
+        Int('entropy',
+            label=_('Entropy'),
+            doc=_('Number of entropy bits'),
+            required=False,
+        ),
+    )
+
+    has_output = (
+        output.summary,
+    )
+
+    def execute(self, *keys, **options):
+        pwd_length = options.get('length')
```

Comment:
You can set default value in `get` function itself. 

`options.get('entropy', 0)`